docs/tutorial.rst

```diff
@@ -306,17 +318,303 @@ trees. In this example we incorrectly inferred that 0 coalesces with the
 ancestor of nodes 2, 3, 4 and 5 before 1.
 
 
-.. todo::
+************
+Data example
+************
+
+Inputting real data for inference is similar in principle to the examples above:
+you simply iterate over the sites in your data file, calling :meth:`.SampleData.add_site`
+for each variable site. We do not provide methods to read directly from other formats, as
+these formats are often too complex for automatic conversion. However, many Python
+libraries are available for reading different genomic data formats, and writing a script
+to iterate over the sites in such files is usually a fairly simple task. The example
+below shows how this can be done for a VCF file, using a freely available VCF reading
+library.
+
++++++++++++++
+Reading a VCF
++++++++++++++
+
+A common way to store genetic variation data is as a VCF (Variant Call Format) file.
+The `CYVCF2 library <https://github.com/brentp/cyvcf2>`_ provides a relatively fast
+Python interface for reading VCF files. The following script demonstrates how to infer a
+tree sequence from a simple VCF file. In particular the function below named
+``add_diploid_sites`` illustrates how to iterate over the variants in a ``CYVCF2.VCF``
+object and add them to a ``tsinfer`` sample data file, while printing out the progression
+through the file in number of base pairs. This particular script assumes that the VCF
+describes a set of phased, diploid individuals, and has the ancestral state present in
+the INFO field of the VCF (otherwise it takes the reference allele as the ancestral
+state). For example data, we use a publicly available VCF file of the genetic variants
+from chromosome 24 of ten Norwegian and French house sparrows, *Passer domesticus*
+(thanks to Mark Ravinet for the data file):
 
-    1. Add documentation links for msprime above so we can explain tree
-       sequences there.
+.. code-block:: python
 
-++++++++++++
-Data example
-++++++++++++
+    import cyvcf2
+    import tsinfer
+    import tqdm
+
+    def add_diploid_sites(vcf, samples):
+        """
+        Loop over sites in the vcf and add them to the samples object, reordering the
+        alleles to put the ancestral allele first, if it is available.
+        """
+        progressbar = tqdm.tqdm(total=samples.sequence_length, desc="Read VCF", unit='bp')
+        pos = 0
+        for variant in vcf:
+            if pos and pos == variant.POS:
```

Comment:
Why would pos be False here? I don't get the first condition.

Reply:
We should allow the first `variant.POS` to be 0 without raising an error. Perhaps clearer if I say `if variant.POS != 0 and pos == variant.POS:` ?

Reply:
Well, why not make pos = -1 or None initially rather than 0 then?

I'm not sure that pos can be zero here anyway, VCF is a 1 based coordinate system.

Reply:
Didn't know that VCF was 1-based. In which case this can be nixed.